### PR TITLE
Jetpack Cloud: Add ability to restore

### PR DESCRIPTION
**Pull request by rcoll**
#### Changes proposed in this Pull Request

* Fill in restore flow with unstyled flow
* Hook up "Restore to this point" button to above flow
*  Several tweaks to jsdoc of selectors we use ( and typescript was complaining about )

##### 1. `/backups/:siteSlug` Start Restore
start restore screenshot

##### 2. `/backups/:siteSlug/restore/:restoreId` Confirm Restore
confirm restore screenshot

##### 3. Restore In Progress

restore in progress screenshot

##### 4. Restore Finished
restore finished screenshot

#### Testing instructions

1. Navigate to `/backups/:siteSlug/restore/:restoreId` for a site with backups enables
2. Find a valid backup ( scrolling back in time if necessary ) 
3. Note the date you are backing up to and click "Restore to this point'
4. Verify you are taken to `/backups/:siteSlug/restore/:restoreId` and it matches the screenshot in step 2 above.
5. Click "Confirm Restore"
6. Verify that the page now displays "BackupRestoreQueued"
7. Verify that the page shortly switches to a progress bar matching step 3 above
8. Verify eventually the restore finished and matches the screenshot fo step 4 above
9. Navigate to `https://wordpress.com/activity-log/:siteSlug` for the same site
10. Verify there is a restore to the same date you noted in step 3

### CODE REVIEW

**Comment by cbauerman** [comment]
> Steps aren't working as expected in my local environment, but I think it's because my test site is somehow broken. I created a new one to test this, but now I have to wait a day to have a full backup. So I'll provide more feedback once my new test site is ready to be tested.

what happened that didn't work as expected? Which steps displayed? I am curious because I am just working from the existing ActivityLog code and would love any edge cases that exist

> Besides that, It seems that the DateRangeSelector does nothing. If you change the date from there, the log entries stay the same (I checked this on the React Dev Tools). Also, the DateRangeSelector doesn't seem to work in sync with the left/right arrow controls. If I move back one day the DateRangeSelector stays the same. The same happens if I change the range (the selected date doesn't change). Are those parts supposed to work in sync?

This is a known issue, it will be a separate PR to bring the date picker online

> **(this is outside the scope of this PR, but I think it's worth to mention it)**
> Finally, this functionality relies on the `getBackupAttemptsForDate` function to get a list of full backups for a given date. I think we may be leaving out some entries because `item.activityDate` is stored in UTC. So, for instance, if I publish a new post on 2020-03-01 22:00:00 GMT-4, that event will be stored as 2020-03-02 02:00:00 UTC (notice how we jumped one day forward). Therefore, if I want to see events on 2020-03-01 I won't see the post publication. I believe the solution is as simple as converting the date to your local date before doing the comparison. `new Date(dateFromApiResponse)` should do the trick.

#39887 should fix this issue, which is what is causing the the merge conflict. I am going to rebase from master and then you should try again.

**Comment by rcoll** [comment]
I see a few quirks, but this is fine as of now. Looks like it needs a rebase and one of the tests needs to pass, but otherwise consider it approved and feel free to :ship:

**Comment by rcoll** [comment]
@rcanepa 

> Besides that, It seems that the DateRangeSelector does nothing. If you change the date from there, the log entries stay the same (I checked this on the React Dev Tools). Also, the DateRangeSelector doesn't seem to work in sync with the left/right arrow controls. If I move back one day the DateRangeSelector stays the same. The same happens if I change the range (the selected date doesn't change). Are those parts supposed to work in sync?

There are definitely some issues with that component that we'll get fixed up in other PRs. Mostly, it's basic localization and some timezone stuff that breaks down. Also, if using with a brand new test site with no backups, I'm sure some bizarre things could happen. :-)

File: client/landing/jetpack-cloud/sections/backups/restore/confirm.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	restoreId?: number | string;
+	siteId: number | null;
+}
+
+const BackupRestoreConfirm = ( { restoreId, siteId, onConfirm }: Props ) => (
+	<div>
+		<p>
+			You have chosen to restore site { siteId } to { restoreId }.
+		</p>
+		<Button onClick={ onConfirm }>{ 'Confirm Restore' }</Button>
```
Comment (ChaosExAnima) [inline client/landing/jetpack-cloud/sections/backups/restore/confirm.tsx]: Why is the text here in brackets?

**Pull request merged**